core/domain/rights_manager.py

```diff
@@ -649,8 +652,12 @@ def check_can_edit_activity(
     if role_services.ACTION_EDIT_OWNED_ACTIVITY not in user.actions:
         return False
 
-    if (activity_rights.is_owner(user.user_id) or
-            activity_rights.is_editor(user.user_id)):
+    if (
+        user.user_id and (
```

Comment:
Probably better to have a "if not user.user_id: return False" before this, and leave this as it was, assuming that from this point onward the user needs to have a user_id to do the action.

Ditto below, where applicable.

Reply:
Done, but in some places, it is not applicable because if we do this we will exceed the limit of return statements in a function that is currently 6.